README.md

```diff
@@ -1,40 +1,125 @@
+*    When there is no match, a return value is provided to describe where and why the license does not match.
 
-Clone the Repository and move to the spdx-python-licensematching folder.
+# Installation
 
-Move the files in the normalize_license_text folder.
+As always you should work in a virtualenv or venv. Do the following steps to setup the Python Package in your system.
 
-Execute the command 
 ```bash
-Python normalize.py <File1> <File2>
+git clone https://github.com/anshuldutt21/spdx_python_licensematching.git
+cd spdx_python_licensematching
+pip install -e .
 ```
-## Compare a License and a Template
-License templates are implemented on the basis of the [SPDX License Matching Guidelines](https://spdx.dev/license-list/matching-guidelines/). In order to use the feature  of comparing a license text and a template-
 
-```bash
-cd compare_template_text/
+# Examples
+
+There are some examples for how to use the code in the examples folder in the directory. Go to the Folder and run the following files to see the output.
+
+```python
+python compare_license_texts.py
+```
+
+This file will compare any 2 sample texts which can be passed into the module and return the output.
+
+```python
+python compare_license_template_text.py
+```
+This file parses the sample text and the template provided in the example and matches them producing the desired output.
+
+```python
+python compare_all_ids.py
+```
+This file parses the sample text and matches it against all the license ids and returns matched licenses.
+
+# How To Use
+
+## Compare 2 License Texts
+License Texts are matched based on the [SPDX License Matching Guidelines](https://spdx.dev/license-list/matching-guidelines/). In order to implement License Matching between 2 License Texts, do the following steps.
+
+Input all the Text Files you want to match anywhere in the file system.
+
+To match  TextFile_A and TextFile_B execute the file normalize.py in the following way.
+
+```python
+python normalize_license_text/normalize.py <Path_to_TextFile_A> <Path_to_TextFile_B>
 ```
+After this step you would get an output depicting whether the license Textfiles match or not along with the text describing the mismatch possibilities.
+
+## Compare a License Text with a License Template
+License templates are implemented on the basis of the [SPDX License Matching Guidelines](https://spdx.dev/license-list/matching-guidelines/). In order to use the feature  of comparing a license text and a template, do the following steps.
+
+Input all the Text Files and the Template files anywhere in the file system.
+
+To match TextFile_A and TemplateFile_A execute the file compare_normalized_files.py in the following way.
+
 ```python
-python compare_template_text.py <TextFile> <TemplateFile>
+python compare_template_text.py/compare_template_text.py <Path_to_TextFile_A> <Path_to_TemplateFile_A>
```

Comment:
Can you add some sample licenses in data/ folder? I see only templates are added.